scripts/histmakers/mw_with_mu_eta_pt.py

```diff
@@ -363,22 +373,27 @@ def build_graph(df, dataset):
     if not args.makeMCefficiency:
         dphiMuonMetCut = args.dphiMuonMetCut * np.pi
         df = df.Filter(f"deltaPhiMuonMet > {dphiMuonMetCut}") # pi/4 was found to be a good threshold for signal with mT > 40 GeV
-
+        
     df = df.Define("passMT", f"transverseMass >= {mtw_min}")
 
     if auxiliary_histograms:
-        # utility plot, mt and met, to plot them later
+        # utility plot, mt and met, to plot them later (need eta-pt to make fakes)
         results.append(df.HistoBoost("MET", [axis_met, axis_eta_utilityHist, axis_pt_utilityHist, axis_charge, axis_passIso, axis_passMT], ["MET_corr_rec_pt", "goodMuons_eta0", "goodMuons_pt0", "goodMuons_charge0", "passIso", "passMT", "nominal_weight"]))
         results.append(df.HistoBoost("transverseMass", [axis_mt_fakes, axis_eta_utilityHist, axis_pt_utilityHist, axis_charge, axis_passIso], ["transverseMass", "goodMuons_eta0", "goodMuons_pt0", "goodMuons_charge0", "passIso", "nominal_weight"]))
 
     ## TODO: next part should be improved, there is quite a lot of duplication of what could happen later in the loop
     ## FIXME: should be isW, to include Wtaunu
-    if isWmunu and args.theoryAgnostic:
-        setTheoryAgnosticGraph(df, results, dataset, reco_sel_GF, era, axes, cols, args)
-        if hasattr(dataset, "out_of_acceptance"):
-            # Rename dataset to not overwrite the original one
-            dataset.name = "Bkg"+dataset.name
-        return results, weightsum
+    if isWmunu and args.theoryAgnostic and not hasattr(dataset, "out_of_acceptance")::
```

Comment:
WE treat the out of acceptance as separate data sets and reprocess them?

Reply:
Yes, this was the way it was originally done by David. Here the reason for this string is that the scope attaches the helicity axis for the histograms defined inside, but for the out-of-acceptance this is not needed because we consider it as a single integrated process. One reason to avoid overflow bins to build it later was actually relevant only for the original theory agnostic version, since all systematic histograms needed the signal gen axes too, but at some point this was nearly unfeasible to run, or extremely small

Reply:
In this way we can treat the out of acceptance as any other background in the subsequent steps